Replace Number.MAX_VALUE with undefined.

mfrener opened the pull request:
**This PR introduces a breaking change**
Because of the breaking change, I have pushed version to 1.2.0

Since we receive too many questions & "bug reports" about values set to Number.MAX_VALUE, we will drop this and use undefined instead. Number.MAX_VALUE was used for values that are not available. It was implemented like that to be in-sync with the official TWS Java API, however checking Number.MAX_VALUE is very uncommon in JScript/TS as everyone would expect `undefined` instead.
So this PR removes Number.MAX_VALUE and replaces it with undefined.

Affects:
IBApi decoder: readDoubleMax / readIntMax has been replaced with readDoubleOrUndefined / readIntOrUndefined
IBApi encoder: encoder still supports Number.MAX_VALUE and will continue to do so for some time. This is to avoid that we submit any insane large numbers to TWS if the client code still uses Number.MAX_VALUE to e.g. create an order.
IBApiNext: undefineMax has been removed as not needed anymore.
README: removed info about Number.MAX_VALUE and add note about the breaking change when upgrading from1.1.x to 1.2.x

Tsopic commented on src/api-next/api-next.ts:
```diff
@@ -680,10 +671,10 @@ export class IBApiNext {
-        unrealizedPnL: undefineMax(unrealizedPnL),
-        realizedPnL: undefineMax(realizedPnL),
-        marketValue: undefineMax(value),
+        dailyPnL: dailyPnL,
+        unrealizedPnL: unrealizedPnL,
+        realizedPnL: realizedPnL,
+        marketValue: value,
```
did not verify it, but could be possible that you forgot to change the interface/type definition to allow undefined.

For example this.
dailyPnL? : number

mfrener replied:
The type on IBApiNext are correct already as we did filter MAX_VALUE already, but just noticed that event callbacks in IBApi still need to be adapted:
```
on(
    event: EventName.pnl,
    listener: (
      reqId: number,
      dailyPnL: number,
      unrealizedPnL: number,
      realizedPnL: number
    ) => void
  ): this;
```
needs to become
```
on(
    event: EventName.pnl,
    listener: (
      reqId: number,
      dailyPnL?: number,
      unrealizedPnL?: number,
      realizedPnL?: number
    ) => void
  ): this;
```
will change it

mfrener commented:
Cross-checked all callback signatures against now possibility undefined values and changed it.
jFYI: 99.9% of all cases are old IBGateway versions (i.e. you connect to version that does not return a given values yet: will result in undefined). 
Types where using optional attributes already almost everywhere, was just one with number instead of a ? number.

Also found that our HISTORICAL_NEWS decoder is buggy (is a copy&paste) during cross-check.
Fixed it.

mfrener commented:
IB really sticky with that MAX_VALUE.... so turns out that this is not only a client-side thing, but the Gatway itself actually returning  '1.797...E308' strings. 
![image](https://user-images.githubusercontent.com/16884927/127651192-c7b07015-ec5f-4997-8f06-1a16c5ee5b99.png)
Fixed it by making readInt() / readDouble() to return 0 when token is empty (as of now), but it will return undefined when token is '1.797...E308'.
Only saw this on pnlSingle so far, on values I already changed to optional. 
But in theory we could see undefined everywhere now. Types are fine, all attributes should be optional there.
On callbacks I really do not want to change every argument to optional now because if this, as I think most of it will never be undefined. We will fix this on a case-by-case basis in case someone faces an undefined value, where the callback signature says it shouldn't.

Tsopic reviewed (commented):
Seems ok if it runs.

mfrener commented:
Checked with my app and all utils from /tools, looks good, no more 1.797...E308
So lets merge and try&error some others :D
